autosklearn/ensemble_building/builder.py

```diff
@@ -595,14 +589,19 @@ def requires_loss_update(
         runs : Sequence[Run]
             The runs to process
 
+        metrics: Sequence[Scorer] | None = None
+            The metrics to check for, defaults to builders
+
         Returns
         -------
         list[Run]
             The runs that require a loss to be calculated
         """
+        metrics = metrics if metrics is not None else self.metrics
+
         queue = []
         for run in sorted(runs, key=lambda run: run.recorded_mtimes["ensemble"]):
-            if run.loss == np.inf:
+            if any(metric.name not in run.losses for metric in metrics):
```

Comment:
Maybe a stupid question, but why do we no longer check for `np.inf`?

Reply:
It used to be the indicator for no loss. Now the indicator is the metric is not recorded in `run.losses`.